pkg/tcpip/stack/stack.go

```diff
@@ -850,9 +859,16 @@ func (s *Stack) GetRouteTable() []tcpip.Route {
 
 // AddRoute appends a route to the route table.
 func (s *Stack) AddRoute(route tcpip.Route) {
+	prefix := route.Destination.Prefix()
 	s.route.mu.Lock()
 	defer s.route.mu.Unlock()
-	s.route.mu.table = append(s.route.mu.table, route)
+	// Sort routes in descending order based on the prefix length
+	i := sort.Search(len(s.route.mu.table), func(j int) bool {
+		return s.route.mu.table[j].Destination.Prefix() <= prefix
```

Comment:
Do we need a tie break in the equal prefix length case?

Reply:
in practice this shouldn't make a difference since an address cannot match with two different subnets with the same prefix length.

However, we probably should make the ordering deterministic.

Reply:
I have added a function Compare which compares the routes based on the destination mask, and address. To make the ordering deterministic, when the destination is unique.